packages/bpk-component-infinite-scroll/src/withInfiniteScroll-test.flow.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow strict */
+
+import React from 'react';
+
+import withInfiniteScroll from './withInfiniteScroll';
+import { ArrayDataSource } from './DataSource';
+
+const elementsArray = [];
+
+for (let i = 0; i < 5; i += 1) {
+  elementsArray.push(`Element ${i}`);
+}
+
+type ListProps = {
+  elements: Array<any>,
+  'aria-label': string,
+  onClick?: ?() => void,
+};
+
+const List = ({ elements, ...rest }: ListProps) => (
+  <div id="list" {...rest}>
+    {elements.forEach(element => (
+      <div key={element}>{element}</div>
+    ))}
+  </div>
+);
+
+List.defaultProps = {
+  onClick: null,
+};
+
+const InfiniteList = withInfiniteScroll(List);
+
+(() => (
+  <React.Fragment>
+    {/* $ExpectError (aria-label is required) */}
+    <InfiniteList dataSource={new ArrayDataSource([])} />
+
+    {/* $ExpectError (dataSource is required) */}
+    <InfiniteList aria-label="infinite list" />
+
+    {/* $ExpectError (incompatible type) */}
+    <InfiniteList dataSource={new ArrayDataSource([])} aria-label={null} />
+
+    {/* Test default props */}
+    <InfiniteList
+      dataSource={new ArrayDataSource([])}
+      aria-label="infinite list"
+    />
```

Comment:
I'm a bit confused as to how this tests default props. I get that when this is rendered the default props will be added, but what are we doing to verify that in this test file?

Reply:
The idea is, if default props are not working we will get a flow error here, so as long as we don't get any error here it means flow is considering default props correctly, or so I hope 😅

Reply:
Ahh gotcha